packages/rust-dprint-plugin-typescript/src/main.rs

```diff
@@ -0,0 +1,189 @@
+            true
+        } else {
+            false
+        }
+    } else {
+        false
+    }
+}
+
+fn check_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path).unwrap();
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error checking: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    if not_formatted_files.is_empty() {
+        Ok(())
+    } else {
+        let f = if not_formatted_files.len() == 1 {
+            "file"
+        } else {
+            "files"
+        };
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("Found {} not formatted {}", not_formatted_files.len(), f,),
+        ))
+    }
+}
+
+fn format_source_files(config: Configuration, paths: Vec<PathBuf>) -> Result<(), std::io::Error> {
+    let mut not_formatted_files = vec![];
+
+    for file_path in paths {
+        let file_path_str = file_path.to_string_lossy();
+        let file_contents = fs::read_to_string(&file_path)?;
+
+        match dprint::format_text(&file_path_str, &file_contents, &config) {
+            Ok(None) => {
+                // nothing to format, pass
+            }
+            Ok(Some(formatted_text)) => {
+                if formatted_text != file_contents {
+                    println!("{}", file_path_str);
+                    fs::write(&file_path, formatted_text)?;
+                    not_formatted_files.push(file_path);
+                }
+            }
+            Err(e) => {
+                eprintln!("Error formatting: {}", &file_path_str);
+                eprintln!("   {}", e);
+            }
+        }
+    }
+
+    let f = if not_formatted_files.len() == 1 {
+        "file"
+    } else {
+        "files"
+    };
+    println!("Formatted {} {}", not_formatted_files.len(), f);
+    Ok(())
+}
+
+pub fn format(
+    config: dprint::configuration::Configuration,
+    target_files: Vec<PathBuf>,
+    check: bool,
+) -> Result<(), std::io::Error> {
+    if check {
+        check_source_files(config, target_files)?;
+    } else {
+        format_source_files(config, target_files)?;
+    }
+    Ok(())
+}
+
+fn create_cli_parser<'a, 'b>() -> clap::App<'a, 'b> {
+    App::new("dprint")
+        .about("Format source files")
+        .long_about(
+            "Auto-format JavaScript/TypeScript source code.
+  
+  dprint myfile1.ts myfile2.ts
```

Comment:
What's the command you guys are going to run in deno to format all the files without specifying them individually? Should we add an example of that here? (I'm not experienced on this topic)

Reply:
So `deno fmt` walks current directory recursively and then checks if they're matching supported extension in `is_supported` function. 

However for our internal code we're actually using `git ls-files` to get complete list of files to format and pass them to `prettier`.